### PR TITLE
feat: merge spec-cycle task delivery loops

### LINKED ISSUE #487: [Loops] Bundled orchestrate-tasks schema rejects approved Goal completion
## Summary

The bundled `orchestrate-tasks` Loop cannot reliably settle an approved successful Goal result because its output schema conflicts with the Goal runtime's authoritative terminal status.

The bundle requires `status` to be `completed|blocked` and requires `summary`. The Goal runtime overwrites the candidate status with its durable Goal status, which is `complete|blocked`. For an approved plain-text/report-tool terminal result with no structured candidate, the runtime creates only `{"status":"complete"}`. Validation therefore reports both:

```text
missing property 'summary'
value must be one of 'completed', 'blocked'
```

In the observed beta.20 run, the final Goal turn was approved and all six downstream worker sessions had stopped, but the bundled action was still marked `invalid_output`.

## CompozyOS version

`compozy 0.3.0-beta.20` (release commit `0cade6fcddc48ddd08182ec5ffcca8ab7d82e244`)

## Operating system

macOS (Apple Silicon)

## Area

- Core / daemon
- Loops & tasks
- Extensions / MCP / skills

## ACP agent

Codex

## Model / provider

`gpt-5.6-sol` / Codex

## Steps to reproduce

Use a fresh temporary workspace with the bundled `spec-cycle` resources and a small neutral valid task suite:

1. Confirm the shipped contract:

```bash
compozy loop inspect --name orchestrate-tasks --workspace <workspace> -o json \
  | jq '.loop.definition.graph.nodes[]
      | select(.id == "orchestrate")
      | .params.output_schema'
```

Expected inspection output includes:

```json
{
  "required": ["status", "summary"],
  "properties": {
    "status": {"enum": ["completed", "blocked"]},
    "summary": {"type": "string"}
  }
}
```

2. Run the Loop and let the neutral tasks satisfy the command judge so the terminal Goal turn is approved.
3. Inspect the final turn and generation output:

```bash
compozy loop turns --run <run-id> --workspace <workspace> -o json \
  | jq '.turns[-1] | {generation, turn, result_status, verdict_outcome, binding_epoch}'

compozy loop status --run-id <run-id> --workspace <workspace> -o json \
  | jq '.generations[0].outputs[]
      | select(.node_id == "orchestrate")
      | {status, output_ref}'
```

Sanitized observation:

```text
final turn: result_status=completed, verdict_outcome=approved
node output: status=failed, code=invalid_output
cause: missing property 'summary'; status not in completed|blocked
```

## What happened, and what did you expect?

**Actual:** A Goal result accepted by the judge fails the bundle's output schema after the Goal reaches its terminal boundary. The Loop then enters recovery instead of settling successfully.

**Expected:** The bundled definition and Goal runtime must use one terminal-status vocabulary and one durable structured-result path. An approved orchestrator result should satisfy the shipped schema and settle the action.

## Impact

- The public bundled Loop can report failure after completing its actual orchestration work.
- Recovery may redo or detach already completed downstream work.
- The error asks the agent to return a valid object after the continuous Goal session has already reached a terminal boundary, so the suggested recovery is not actionable in-place.

## Likely subsystem / root-cause hypothesis

**Hypothesis, directly supported by the beta.20 bundle and runtime source:** this is a shipped contract mismatch plus an incomplete plain-text fallback.

- [`extensions/spec-cycle/loops/orchestrate-tasks/loop.yaml:85-94`](https://github.com/compozy/compozy/blob/0cade6fcddc48ddd08182ec5ffcca8ab7d82e244/extensions/spec-cycle/loops/orchestrate-tasks/loop.yaml#L85-L94) requires `status`, `summary`, and the enum `completed|blocked`.
- [`internal/loop/goal/result.go:12-35`](https://github.com/compozy/compozy/blob/0cade6fcddc48ddd08182ec5ffcca8ab7d82e244/internal/loop/goal/result.go#L12-L35) overwrites `status` with the authoritative Goal status and, when there is no structured candidate, emits no other fields.
- [`internal/loop/goal/executor_test.go:797-833`](https://github.com/compozy/compozy/blob/0cade6fcddc48ddd08182ec5ffcca8ab7d82e244/internal/loop/goal/executor_test.go#L797-L833) codifies the durable successful Goal status as `complete`, not `completed`.

## Recommended repair

- Align the bundled schema and prompt with the runtime's closed Goal status vocabulary (`complete|blocked`), or add an explicit, tested mapping at the Goal action boundary. Avoid maintaining two implicit spellings.
- Provide a durable schema-compliant result path for Goal reporting. Either let the Goal report tool carry and persist the full output payload, or make fields such as `summary` optional when the runtime can only synthesize status from plain text.
- Validate the final structured payload before closing the continuous binding. If repair is possible, issue a bounded schema-repair turn on the same binding rather than failing the generation after approval.

## Regression tests / acceptance criteria

1. Load the real bundled `orchestrate-tasks/loop.yaml` in a contract test; do not duplicate its schema in the test.
2. Complete a neutral orchestration with an approved plain-text/report-tool terminal result and assert the action succeeds with a schema-valid payload.
3. Complete with an explicit structured payload containing status, summary, and tasks; assert all fields survive authoritative status projection.
4. Exercise `blocked` through the same path.
5. If a malformed payload is repairable, assert one bounded repair turn occurs before session closure and the run settles without starting a new generation.
6. Verify extension inventory, official Compozy Loop skill/docs, CLI/HTTP/UDS output, and the bundled definition stay in sync.

## Related issues

- #480 is about cleanup payload idempotence during settlement; it does not cover this bundled schema mismatch.
- #472 proposes a new reattempt policy; it does not make an approved terminal payload schema-valid.
- #451 is an upgrade/boot-resumption problem, not a same-version shipped Loop contract mismatch.

## Logs

```shell
invalid_output: loop: action output schema validation failed
- at '': missing property 'summary'
- at '/status': value must be one of 'completed', 'blocked'
```

## Final checks

- [x] I searched existing issues and pull requests; I did not find an adequate duplicate.
- [x] I can reproduce this on the latest release, `0.3.0-beta.20`.

### PR DESCRIPTION
**Pull request by pedronauck**
## What & why

Fixes #487.

This PR replaces the two spec-cycle task-delivery Loops with one mode-selectable `implement-tasks` Loop. The default remains the existing per-task delivery path, so current callers that only provide `slug`, `implementer`, or `auto_commit` keep their behavior and do not receive a new required prompt. Operators who want one continuous conductor can set `mode=orchestrated` instead of choosing a second catalog entry.

### One task-delivery Loop

- Keeps `implement-tasks` as the stable Loop name and the only task-delivery entry.
- Adds `mode` with `per-task` as the default and `orchestrated` as the alternative.
- Preserves dependency-ordered, sequential, isolated task sessions in per-task mode.
- Loads the same task graph in both modes, routes each fan-out item by task category, collects the staged results, and only then routes to the selected delivery endpoint.
- Marks the unselected delivery path as `not_taken` and settles both modes as `done`.
- Keeps schedule starts and the shared `concurrency: forbid` domain.
- Hard-deletes the standalone `orchestrate-tasks` Loop and all current catalog, fixture, docs, skill-reference, and QA surfaces that treated it as a separate capability.

The graph stages items before the orchestrated Goal because route skip propagation is scoped to the fan-out item. Routing the whole graph before fan-out leaves the shared collector pending. The merged topology keeps the engine unchanged while giving both modes one legal join point.

### Category-specific runtime selection

The run contract adds four optional runtime inputs:

- `orchestrator_runtime`
- `backend_runtime`
- `frontend_runtime`
- `default_runtime`

Task category is the task file's `type` frontmatter. `backend` uses `backend_runtime`, `frontend` uses `frontend_runtime`, and every other value uses `default_runtime` through `code_implementer`. A task's own frontmatter `runtime` remains more specific and therefore still wins over the run input under ADR-001. Empty inputs fall through to existing agent/config/rule resolution. `orchestrator_runtime` is inert in per-task mode.

Orchestrated workers now preserve the full runtime choice. The public spawn contract carries provider, model, reasoning effort, and speed across CLI, HTTP/UDS core, native tools, OpenAPI, generated TypeScript, docs, and digest-locked tool metadata. The conductor uses public `compozy spawn`, blocking `compozy session prompt`, on-disk task-frontmatter proof, and `compozy session stop`; it does not edit production code itself.

### Bundled orchestrator agent

The spec-cycle extension now bundles an `orchestrator` agent and uses it as the conductor default. Its contract is delegation-only: one bounded worker per task, independent disk proof, one corrective turn at most, and teardown on every path. The existing `cy-orchestrate-tasks` skill remains one of the nine bundled skills and now documents category mapping plus full runtime dispatch.

### Goal contract repair and defect-class guards

The broken standalone Loop required `completed|blocked`, while the Goal runtime's public terminal vocabulary is `complete|blocked`. The runtime correctly projected `complete`, so the authored schema could never validate, and plain-text approvals also lacked a required summary.

This PR fixes the bundle rather than changing the Goal public contract:

- The merged Goal schema accepts `complete|blocked` and only requires the authoritative status.
- The incident-postmortem demo uses the same correct vocabulary.
- The linter now rejects Goal output schemas whose status enum omits either `complete` or `blocked`.
- Goal prompts receive the exact output-schema contract through the same shared prompt helper used by run-agent.
- Contract tests load the real bundled YAML, prove approved plain text is schema-valid, preserve explicit structured fields under authoritative status projection, and exercise the blocked path.

### Hard cut and public surfaces

- Bumps the bundled spec-cycle extension to `0.5.0`.
- Removes the old site example and Web mock fixture.
- Updates the site examples, Loop catalog, configuration examples, session/spawn docs, official Compozy skill references, generated OpenAPI types, and bundled marketplace inventory.
- Adds a release note that calls out inert operator-side `[loops.inputs.orchestrate-tasks]` config blocks after the hard cut.
- Retires the old QA scenario as `skipped`, moves coverage to a content-addressed orchestrated-mode scenario, and updates affected journeys, personas, charters, and tracker claims. Dated historical reports are untouched.

The review-and-fix Loop, Goal runtime vocabulary, template language, graph engine semantics, Batuta behavior, and config schema are unchanged.

## How you verified it

Focused checks completed before opening this draft:

- `make build` — passed (`CodegenCheck` and Go binary build).
- `go test -tags=integration ./internal/daemon -run '^TestDaemonE2EImplementTasksShouldCompleteTaskJourney' -count=1` — both default per-task and orchestrated journeys passed.
- Focused extension/Goal/runtime/API/CLI/native-tool/acpmock Go suites — passed.
- `bunx turbo run test --filter=compozy-web -- ...` for Loop adapters, hooks, catalog, detail, and run form — 75/75 passed.
- `bunx turbo run test --filter=@compozy/site -- lib/__tests__/runtime-docs-truth.test.ts` — 15/15 passed.
- `bunx turbo run test --filter=@compozy/site -- lib/__tests__/marketplace-catalog.test.tsx lib/__tests__/public-route-metadata.test.ts lib/__tests__/public-search-index.test.ts` — 58/58 passed.
- `make gate` — passed for the current tree before commit/push; codegen, zero-warning Go lint, affected Go tests under `-race`, and affected Turbo lanes are recorded in `.cache/gate/`.

The targeted living-docs QA round is in progress in a fresh isolated lab. It covers the Loop catalog, typed input form, both execution modes, dry-run, extension leave/return lifecycle, public examples, and bundled marketplace inventory. The PR stays draft until that walk and the exact-head required CI checks are green.

### Public behavior

- **CLI:** `compozy spawn` adds `--reasoning-effort` and `--speed`; `compozy loop run implement-tasks` exposes the mode, orchestrator, and four optional runtime inputs through the existing generic input surface.
- **HTTP/UDS:** the existing agent spawn request accepts reasoning effort and speed. No route changes.
- **Native tools:** session orchestration spawn input and descriptor now carry reasoning effort and speed; the catalog digest is regenerated.
- **Web:** the existing schema-driven run form renders the mode enum, orchestrator agent selector, and runtime controls. No custom UI or unsupported control was added.
- **Config:** no new config key or parser. Existing `loops.inputs.implement-tasks.*` persistence accepts the new inputs. The deleted Loop's dynamic input block becomes inert and is documented as such.
- **Docs:** runnable examples, generated CLI/API references, Loop catalog/config/session docs, marketplace inventory, and the official Compozy skill references are updated.

### Compozy Impact Audit

- **Native tools:** changed the existing spawned-session input/descriptor for reasoning effort and speed; updated its schema digest and tests. Loop run inputs continue through the existing generic native tool contract.
- **Extensibility and hooks:** spec-cycle inventory changes from three Loops/three agents to two Loops/four agents; `cy-orchestrate-tasks` remains the conductor capability; no hook, MCP sidecar, resource kind, registry protocol, or bridge SDK changes. Config lifecycle uses the existing dynamic Loop-input namespace.
- **Workspace data isolation:** no new persisted datum or ownership scope. Loop inputs remain run-scoped inside the existing workspace-owned run; spawned worker sessions retain parent/workspace identity. CLI, HTTP/UDS, core, store, SSE, cache, and event ownership paths are unchanged.
- **Official Compozy skill:** updated `skills/compozy/` Loop and session-runtime references for the merged mode and full spawn runtime fields; tool IDs and command paths stay stable.

### Breaking change

`orchestrate-tasks` is intentionally removed with no alias or compatibility shim. Historical runs remain readable through existing pinned definition snapshots. Current callers must use `implement-tasks` with `mode=orchestrated`.

---

- [x] `make gate` passes locally; this PR is delivered only after its required CI checks are green
- [x] New or changed behavior is covered by tests, including both mode routes and the #487 regression contract
- [x] Codex (GPT-5.6, high reasoning) authored the implementation; focused suites and the local gate independently verified it

## Summary by CodeRabbit

* **New Features**
  * Added direct and orchestrated task-implementation modes.
  * Added backend, frontend, and default execution paths.
  * Added configurable runtime, reasoning-effort, and speed options for spawned agents.
  * Added CLI support for `--reasoning-effort` and `--speed`.
* **Improvements**
  * Enhanced completion handling, output validation, worker cleanup, and session identity propagation.
  * Consolidated orchestration into the expanded implementation workflow.
* **Bug Fixes**
  * Improved action and recovery prompt error handling.
  * Added validation requiring complete and blocked goal statuses.

### CODE REVIEW

File: internal/loop/goal/templates.go [hunk 1]
```diff
@@ -115,7 +116,14 @@ func renderWorkPrompt(segment *segmentState, turn int, kind string) string {
 			"Use the Goal report tool only for an explicit complete or evidenced blocked boundary; " +
 			"reporting does not end the current prompt.",
 	)
-	return prompt.String()
+	return loop.ActionPromptWithOutputContract(prompt.String(), dereferenceSchema(segment.params.OutputSchema))
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply internal/loop/goal/templates.go:119]: Verified against the runtime and the accepted brief: this finding does not apply. R6 explicitly requires the Goal work prompt to expose the authored output_schema with exact parity to run-agent, and the canonical test asserts ActionPromptWithOutputContract equality. Managed work-turn responses are not schema-validated per turn; validation occurs only after the authoritative terminal Goal boundary in executor.go. Removing the contract from work prompts would reopen issue #487 and violate R6. I am retaining the required behavior.
